### PR TITLE
Rename SimulatorEnclaveProvider - "None" attestation protocol, phase 1

### PR DESCRIPTION
**Pull request by johnnypham**
**Summary of feature**
Currently, VBS enclaves are supported on-prem and SGX enclaves on Azure VMs. VBS support for Azure is in the works but strong attestation is not possible on Azure VMs due to restrictions against accessing the host machine's Trusted Platform Module (TPM). 
A new attestation protocol called "None" will be allowed in the connection string, allowing users to forgo enclave attestation when using VBS enclaves: `Attestation Protocol = None;` 

**Phase 1 changes**
The existing `SimulatorEnclaveProvider` performs no attestation and only derives the shared secret required to set up a secure enclave session, which is exactly what is needed. This PR renames it to the proposed `NoneAttestationEnclaveProvider` and the file can now be shared across netcore and netfx.

The simulator can be built with 
```bash
msbuild -p:Configuration="Release" -p:BuildSimulator=true
````

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/NoneAttestationEnclaveProvider.cs [hunk 1]
```diff
@@ -29,18 +25,15 @@ internal override void GetEnclaveSession(EnclaveSessionParameters enclaveSession
         // Gets the information that SqlClient subsequently uses to initiate the process of attesting the enclave and to establish a secure session with the enclave.
         internal override SqlEnclaveAttestationParameters GetAttestationParameters(string attestationUrl, byte[] customData, int customDataLength)
         {
-            // The key derivation function and hash algorithm name are specified when key derivation is performed
-            ECDiffieHellman clientDHKey = ECDiffieHellman.Create();
-            clientDHKey.KeySize = 384;
-
-            return new SqlEnclaveAttestationParameters(2, new byte[] { }, clientDHKey);
+            ECDiffieHellman clientDHKey = KeyConverter.CreateECDiffieHellman(DiffieHellmanKeySize);
```
Comment (johnnypham) [inline src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/NoneAttestationEnclaveProvider.cs:28]: This change is required to share the file. See #1022

Comment (Wraith2) [reply src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/NoneAttestationEnclaveProvider.cs:28]: Could `new byte[] { }` be `Array.Empty<byte>()` or is it not available in all supported targets?

Comment (johnnypham) [reply src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/NoneAttestationEnclaveProvider.cs:28]: I think it'll work, I'll let the CI determine that 😄

**Review by Kaur-Parminder — APPROVED** [review]

File: src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj [hunk 2]
```diff
@@ -629,9 +632,6 @@
     <Compile Include="Microsoft\Data\SqlClient\Server\sqlser.cs" />
     <Compile Include="Microsoft\Data\Sql\SqlGenericUtil.cs" />
   </ItemGroup>
-  <ItemGroup Condition="'$(BuildSimulator)' == 'true'">
-    <Compile Include="Microsoft\Data\SqlClient\SimulatorEnclaveProvider.cs" />
-  </ItemGroup>
```
Comment (JRahnama) [inline src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj:634]: is removing the condition intentional?

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/NoneAttestationEnclaveProvider.cs [hunk 3]
```diff
@@ -3,21 +3,17 @@
...
-using System.Collections.Concurrent;
 
 namespace Microsoft.Data.SqlClient
 {
-    internal class SimulatorEnclaveProvider : EnclaveProviderBase
+    internal class NoneAttestationEnclaveProvider : EnclaveProviderBase
     {
         private static readonly int EnclaveSessionHandleSize = 8;
+        private const int DiffieHellmanKeySize = 384;
+        private const int NoneAttestationProtocolId = 2;
```
Comment (DavoudEshtehari) [inline src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/NoneAttestationEnclaveProvider.cs:16]: This value equals to `SqlConnectionAttestationProtocol.SIM` which builds only with `ENCLAVE_SIMULATOR=true` at build time. Could you add some notes to explain the source of the value **2**?

**Review by DavoudEshtehari — APPROVED** [review]

File: src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj (hunk 2 above)
Comment (johnnypham) [reply src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj:634]: Yes, in phase 2, the newly named  `NoneAttestationEnclaveProvider` will now actually be used within the driver when `Attestation Protocol = None;` is specified. Before it was only used for testing purposes when the simulator flag was set.

**Review by JRahnama — APPROVED** [review]

**Pull request merged**